### PR TITLE
Sagas/node

### PR DESCRIPTION
**Pull request by ramirobg94**

### CODE REVIEW

File: src/state/nodes/sagas/createNode/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  const childrenNodeDefs = yield select(state =>
+    surveySelectors.getNodeDefChildren(state, parentNodeDef),
+  );
+
+  if (childrenNodeDefs?.length <= 0) {
+    return [];
+  }
+
+  const nodes = yield all(
```
Comment (SteRiccio) [inline src/state/nodes/sagas/createNode/index.js]: What is this? Why don't you use arena core to create nodes? What about nodes in single entities and what about applying default values to that nodes?

File: src/state/nodes/sagas/createNode/index.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  const node = yield call(_createNode, {
+    record,
+    parentNode,
+    nodeDef,
+    survey,
+  });
+
+  yield put(nodesActions.setNode({node}));
+
+  if (nodeDef.type === 'entity') {
```
Comment (SteRiccio) [inline src/state/nodes/sagas/createNode/index.js]: Please use the NodeDefType in core

File: src/state/nodes/actionCreators.js [hunk 3]
```diff
@@ -4,7 +4,10 @@ import types from './actionTypes';
 
 const {nodes} = createActions({
   [types.createNode$]: ({nodeDef}) => ({nodeDef}),
-  [types.updateNode$]: ({updatedNode}) => ({updatedNode}),
+  [types.updateNode$]: ({updatedNode, shouldClose}) => ({
+    updatedNode,
+    shouldClose,
```
Comment (SteRiccio) [inline src/state/nodes/actionCreators.js]: Should close what? Should close the node editor?

**Review by SteRiccio — CHANGES_REQUESTED** [review]

File: src/state/nodes/sagas/createNodeAndDescendants/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  const recordWithNodes = {...record, nodes: {...(recordNodes || {})}};
+
+  const updateRecord = RecordNodesUpdater.addNodeAndDescendants({
+    survey,
+    record: recordWithNodes,
+    parentNode,
+    nodeDef,
+  });
+
+  // TODO Add surveyUuid into CORE and fix dates
```
Comment (ramirobg94) [inline src/state/nodes/sagas/createNodeAndDescendants/index.js]: in core, because they are without timezone in core

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**